packages/parcel-transformer/src/transformer.tsx

```diff
@@ -64,54 +67,63 @@ export default new Transformer<UserlandOpts>({
 
   async transform({ asset, config }) {
     const ast = await asset.getAST();
-    if (!asset.isSource || !ast) {
-      // We will only recieve ASTs for assets we're interested in.
+    if (!ast) {
```

Comment:
I believe this will bail out of the transform entirely when the asset is not source or we are extracting, since you check for the presence of the ast from `parse`. These scenarios should all be tested in code with the programmatic api...